### PR TITLE
Disassociate elastic IP address when detaching Internet Gateway

### PR DESCRIPTION
**Pull request by youyangl**
According to AWS tutorial:
https://docs.aws.amazon.com/vpc/latest/userguide/VPC_Internet_Gateway.html#detach-igw The elastic ip should be disassociated before detaching vpc internet gateway.

### CODE REVIEW

**Review by elton-d — APPROVED** [review]

**Comment by justinsb** [comment]
Thanks for fixing!

/approve
/lgtm

**Comment by justinsb** [comment]
Thanks!

/approve
/lgtm

File: aws-janitor/resources/internet_gateways.go [hunk 1]
```diff
@@ -77,6 +77,37 @@ func (InternetGateways) MarkAndSweep(opts Options, set *Set) error {
 				break
 			}
 
+			var publicIPsToRelease []*string
+
+			pageFunc := func(page *ec2.DescribeNetworkInterfacesOutput, _ bool) bool {
+				for _, eni := range page.NetworkInterfaces {
+					publicIPsToRelease = append(publicIPsToRelease, eni.Association.PublicIp)
```
Comment (chemikadze) [inline aws-janitor/resources/internet_gateways.go:84]: I believe association can be nil, causing cleanup to panic in certain cases. This also made difficult by code in main() that appears to hide panic logging as it calls explicit os.Exit.

Also, deleting network interfaces should disassociate automatically, so likely we don't need this change - I'm suspecting that ENIs are held by untagged ELBs.

**Pull request merged**